python/ray/data/tests/test_arrow_block.py

```diff
@@ -1,15 +1,251 @@
+import os
+import sys
 import types
+from tempfile import TemporaryDirectory
+from typing import Union
 
 import numpy as np
 import pyarrow as pa
 import pytest
+from pyarrow import parquet as pq
 
 import ray
 from ray._private.test_utils import run_string_as_driver
+from ray.air.util.tensor_extensions.arrow import ArrowTensorArray
+from ray.data import DataContext
 from ray.data._internal.arrow_block import ArrowBlockAccessor
+from ray.data._internal.arrow_ops.transform_pyarrow import combine_chunked_array
+from ray.data._internal.util import GiB, MiB
 from ray.data.extensions.object_extension import _object_extension_type_allowed
 
 
+@pytest.fixture(scope="module")
+def parquet_dataset_single_column_gt_2gb():
+    chunk_size = 256 * MiB
+    num_chunks = 10
+
+    total_column_size = chunk_size * 10  # ~2.5 GiB
+
+    with TemporaryDirectory() as tmp_dir:
+        dataset_path = f"{tmp_dir}/large_parquet_chunk_{chunk_size}"
+
+        # Create directory
+        os.mkdir(dataset_path)
+
+        for i in range(num_chunks):
+            chunk = b"a" * chunk_size
+
+            d = {"id": [i], "bin": [chunk]}
+            t = pa.Table.from_pydict(d)
+
+            print(f">>> Table schema: {t.schema} (size={sys.getsizeof(t)})")
+
+            filepath = f"{dataset_path}/chunk_{i}.parquet"
+            pq.write_table(t, filepath)
+
+            print(f">>> Created a chunk #{i}")
+
+        print(f">>> Created dataset at {dataset_path}")
+
+        yield dataset_path, num_chunks, total_column_size
+
+        print(f">>> Cleaning up dataset at {dataset_path}")
+
+
+@pytest.fixture(scope="module")
+def binary_dataset_single_file_gt_2gb():
+    total_size = int(2.1 * GiB)
+    chunk_size = 256 * MiB
+    num_chunks = total_size // chunk_size
+    remainder = total_size % chunk_size
+
+    with TemporaryDirectory() as tmp_dir:
+        dataset_path = f"{tmp_dir}/binary_dataset_gt_2gb_single_file"
+
+        # Create directory
+        os.mkdir(dataset_path)
+
+        with open(f"{dataset_path}/chunk.bin", "wb") as f:
+            for i in range(num_chunks):
+                f.write(b"a" * chunk_size)
+
+                print(f">>> Written chunk #{i}")
+
+            if remainder:
+                f.write(b"a" * remainder)
+
+        print(f">>> Wrote chunked dataset at: {dataset_path}")
+
+        yield dataset_path, total_size
+
+        print(f">>> Cleaning up dataset: {dataset_path}")
+
+
+@pytest.mark.parametrize(
+    "col_name",
+    [
+        "bytes",
+        # TODO fix numpy conversion
```

Comment:
I assume this will be fixed in this PR?

Reply:
Unfortunately numpy does not support reading strings larger than 2 Gb